scripts/run-test.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+function checkForConditionalRun {
+  if [ "$TRAVIS" == "ci" ];
+  then
+    echo "Checking if this should be conditionally run.."
+    result=$(git diff --name-only HEAD | grep azurerm/)
+    if [ "$result" = "" ];
+    then
+      echo "No changes committed to ./azurerm - nothing to lint - exiting"
+      exit 0
+    fi
+  fi
+}
+
+function runTests {
+  echo "==> Running Unit Tests..."
+  go test -i $TEST || exit 1
+	go test -v $TEST "$TESTARGS" -timeout=30s -parallel=4
```

Comment:
globbing / splitting protection
```suggestion
	go test -v "$TEST" "$TESTARGS" -timeout=30s -parallel=4
```

Reply:
in this case TEST will always have a value, (since it's defaulted in the Gnumakefile) - whereas TESTARGS won't - so I don't think that's needed?

Reply:
I'm a fan of `shellcheck`, so this habit is ingrained 😄- I don't think it's essential here though.